File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mp;
        int size = nums.size();
        int maxCount = 0;
        int maxElement = 0;
        for(int n: nums){
            if(++mp[n] > (size/2)){
                if(maxCount < mp[n]){
                    return maxElement = n;
                }
            }
        }
        return maxElement;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mp = {}
        maxCount, majorityElement = 0, 0

        for num in nums:
            mp[num] = mp.get(num, 0)+1
            if maxCount < mp[num]:
                maxCount = mp[num]
                majorityElement = num

        if mp[majorityElement] > len(nums)/2:
            return majorityElement
        return -1

        
        